docs: added publishable API keys documentation pages (#2972)
medusajs/medusa: docs/content/advanced/backend/publishable-api-keys/index.md

# Publishable API Keys Overview

> In this document, you’ll learn about what Publishable API Keys are and how they work.

Reviewer: suggestion(non-blocking):
```suggestion
In this document, you’ll learn about Publishable API Keys and their usage.
```

## Introduction

While using Medusa’s APIs, you might have to pass some query parameters for certain resources with every or most requests.

Taking Sales Channels as an example, you have to pass the Sales Channel’s ID as a query parameter to all the necessary endpoints, such as the List Products endpoint.

> As this approach can be error-prone, Publishable API Keys solve this problem.

Reviewer: suggestion(non-blocking):
```suggestion
This is a tedious and error-prone process, which is why we introduce Publishable API Keys.
```

- `id`: The ID of the publishable API key. This is the API key you’ll use in your API requests.
- `created_by`: The ID of the user that created this API key.
- `revoked_by`: The ID of the user that revoked this API key. A revoked publishable API key cannot be used in requests.

---

## Relation to Other Entities

### Sales Channels

A publishable API key can be associated with more than one sales channel, and a sales channel can be associated with more than one publishable API key.

The relation is represented by the entity `PublishableApiKeySalesChannel`.

---

## Using Publishable API Keys in Requests

:::note

Publishable API keys are only for client-side use. They can be publicly accessible in your code, as they are not authorized for the Admin API.

:::

### Using Medusa JS Client

When using [Medusa’s JS Client](../../../js-client/overview.md), you can pass it to the client only once when you create the instance of the client:

```ts
const medusa = new Medusa({
  maxRetries: 3,
  baseUrl: "https://api.example.com",
  publishableApiKey,
})
```

> Then, the API key will be passed in the header parameter `x-publishable-api-key` of every request.

Reviewer: suggestion(non-blocking):
```suggestion
This will add the API key as a header, `x-publishable-api-key`, on all requests.
```
